src/Subscriptions/StorageManager.php

```diff
@@ -110,8 +111,43 @@ public function storeSubscriber(Subscriber $subscriber, string $topic): void
         }
     }
 
+    /**
+     * Delete subscriber.
+     *
+     * @param  string  $channel
+     * @return \Nuwave\Lighthouse\Subscriptions\Subscriber|null
+     */
     public function deleteSubscriber(string $channel): ?Subscriber
     {
-        return $this->cache->pull(self::SUBSCRIBER_KEY.".{$channel}");
+        $subscriber = $this->cache->pull(self::SUBSCRIBER_KEY.".{$channel}");
+
+        if ($subscriber !== null) {
+            $this->removeSubscriberFromTopic($subscriber);
+        }
+
+        return $subscriber;
+    }
+
+    /**
+     * Remove the subscriber from the topic they are subscribed to.
+     *
+     * @param  Subscriber $subscriber
+     */
+    protected function removeSubscriberFromTopic(Subscriber $subscriber)
+    {
+        if (empty($subscriber->topic)) {
+            return;
+        }
+
+        $topic = Collection::make(json_decode($this->cache->get($subscriber->topic), true))
+            ->reject(function ($key) use ($subscriber) {
+                return self::SUBSCRIBER_KEY.".{$key}" === self::SUBSCRIBER_KEY.".{$subscriber->channel}";
+            });
+
```

Comment:
```suggestion

        if ($topic->isEmpty()) {
            $this->cache->forget($subscriber->topic);
            
            return;
        }

```

To prevent storing empty topic, we might want to `forget` empty topics.

Reply:
Good call.